src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -112,11 +117,53 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             );
         }
         try {
+
+            Template template;
+            Map<String, String> useCaseDefaultsMap = Collections.emptyMap();
             XContentParser parser = request.contentParser();
             ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-            Template template = Template.parse(parser);
+            if (useCase != null) {
+                String json = ParseUtils.resourceToString("/" + DefaultUseCases.getSubstitutionReadyFileByUseCaseName(useCase));
+                String defaultsFilePath = DefaultUseCases.getDefaultsFileByUseCaseName(useCase);
+                useCaseDefaultsMap = ParseUtils.parseJsonFileToStringToStringMap("/" + defaultsFilePath);
+
+                if (request.hasContent()) {
+                    try {
+                        Map<String, String> userDefaults = ParseUtils.parseStringToStringMap(parser);
+                        // updates the default params with anything user has given that matches
+                        for (Map.Entry<String, String> userDefaultsEntry : userDefaults.entrySet()) {
+                            if (useCaseDefaultsMap.containsKey(userDefaultsEntry.getKey())) {
```

Comment:
for better readability 
```suggestion
                            if (useCaseDefaultsMap.containsKey(userDefaultsEntry.getKey())) {
                            String key = userDefaultsEntry.getKey();
                            String value = userDefaultsEntry.getValue();
                            
```